### PR TITLE
Migrate msgfees gov props v1

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Description

closes: #1328

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by Taztingo — DISMISSED** [review]

File: x/msgfees/types/msgs_test.go [hunk 1]
```diff
@@ -77,3 +79,298 @@ func TestMsgAssessCustomMsgFeeValidateBasic(t *testing.T) {
...
+	cases := []struct {
+		name     string
+		msg      *MsgAddMsgFeeProposalRequest
+		errorMsg string
+	}{
+		{
+			"Empty type error",
+			NewMsgAddMsgFeeProposalRequest("", sdk.NewCoin("hotdog", sdk.NewInt(10)), "", "", authority),
+			"msg type is empty",
+		},
```
Comment (SpicyLemon) [inline x/msgfees/types/msgs_test.go:95]: **Always** use named parameters when defining test cases. **Always**.
Without them, you can't make any changes to the test case struct without having to change every single test case, even when the test case isn't actually changing. They also make the tests easier to understand lower in the list where you don't have the struct definition on the screen anymore.

File: x/msgfees/types/msgs_test.go [hunk 2]
```diff
@@ -77,3 +79,298 @@ func TestMsgAssessCustomMsgFeeValidateBasic(t *testing.T) {
...
+		{
+			"Invalid fee amounts",
+			NewMsgAddMsgFeeProposalRequest("msgType", sdk.NewCoin("hotdog", sdk.NewInt(0)), "", "", authority),
```
Comment (SpicyLemon) [inline x/msgfees/types/msgs_test.go:98]: Relatedly, these tests are significantly easier to maintain if you don't use the `New...` constructor to make the thing you're testing. When a field is added to a message, and therefore the constructor, you then have to go back through all the tests and update the constructor call. But usually, when adding a field like that, all of the existing unit tests are still valid and wouldn't need to be changed if they didn't use the constructor.

File: x/msgfees/types/msgs_test.go (hunk 1 above)
Comment (nullpointer0x00) [reply x/msgfees/types/msgs_test.go:95]: Dang missed this.  I usually do the named parameters.  Don't know why I missed them here.

File: x/msgfees/types/msgs_test.go (hunk 2 above)
Comment (nullpointer0x00) [reply x/msgfees/types/msgs_test.go:98]: I will change them.  However, I do like having the constructor for that reason of breaking test or anywhere it is used.  It forces me to review the tests and update accordingly or remove ones that might be invalid after the updated constructor.  I was usually just using the fields when I needed to create the object in a state the constructor didn't allow.  I will start using the fields now going forward, just wanted to clarify my reasoning behind it.

Comment (SpicyLemon) [reply x/msgfees/types/msgs_test.go]: If the test itself doesn't break, we shouldn't have to update the code for it.

**Review by SpicyLemon — APPROVED** [review]

**Review by iramiller — APPROVED** [review]

**Pull request merged**